tests/memgraph/test_query_builder.py

```diff
@@ -1340,3 +1340,57 @@ class Follows_test(Relationship, type="FOLLOWS"):
         query_builder.execute()
 
     mock.assert_called_with(expected_query)
+
+
+def test_bfs():
+    bfs_alg = BreadthFirstSearch()
+
+    query_builder = (
+        QueryBuilder()
+        .match()
+        .node(labels="City", name="Zagreb")
+        .to(edge_label="Road", algorithm=bfs_alg)
+        .node(labels="City", name="Paris")
+        .return_()
+    )
+    expected_query = " MATCH (:City {name: 'Zagreb'})-[:Road * BFS]->(:City {name: 'Paris'}) RETURN * "
+
+    with patch.object(Memgraph, "execute_and_fetch", return_value=None) as mock:
+        query_builder.execute()
+
+    mock.assert_called_with(expected_query)
+
+
+def test_bfs_filter_label():
+    bfs_alg = BreadthFirstSearch(condition="e.length <= 200 AND v.name != 'Metz'")
+
+    query_builder = (
+        QueryBuilder()
+        .match()
+        .node(labels="City", name="Paris")
+        .to(edge_label="Road", algorithm=bfs_alg)
+        .node(labels="City", name="Berlin")
+        .return_()
+    )
+
+    expected_query = " MATCH (:City {name: 'Paris'})-[:Road * BFS (e, v | e.length <= 200 AND v.name != 'Metz')]->(:City {name: 'Berlin'}) RETURN * "
+
+    with patch.object(Memgraph, "execute_and_fetch", return_value=None) as mock:
+        query_builder.execute()
+
+    mock.assert_called_with(expected_query)
+
+
+def test_bfs_bounds():
```

Comment:
Add tests with both lower and upper bounds, and test with just lower bound to cover all cases.